std/typecons.d

```diff
@@ -2803,7 +2791,21 @@ Params:
  */
     void opAssign()(T value)
     {
-        _value = value;
+        import std.algorithm.mutation : moveEmplace, move;
+
+        // the lifetime of the value in copy shall be managed by
+        // this Nullable, so we must avoid calling its destructor.
+        auto copy = DontCallDestructorT(value);
+
+        if (_isNull)
+        {
+            // trusted since _value is known to be T.init here.
+            () @trusted { moveEmplace(copy.t, _value.t); }();
+        }
+        else
+        {
+            move(copy.t, _value.t);
```

Comment:
In this branch we know `_value` is initialized so why avoid calling the postblit? It seems to me like this might cause something we don't want, like a `RefCounted` argument having its count drop to zero and deallocating its storage after we assign it to `_value`.

Reply:
Ah wait! The postblit gets called by `auto copy = DontCallDestructorT(value)`.